diff --git a/node_modules/eth-lightwallet/lib/keystore.js b/node_modules/eth-lightwallet/lib/keystore.js
index 423b47c..5ed846c 100644
--- a/node_modules/eth-lightwallet/lib/keystore.js
+++ b/node_modules/eth-lightwallet/lib/keystore.js
@@ -339,7 +339,7 @@ KeyStore.deriveKeyFromPasswordAndSalt = function(password, salt, callback) {
   var logN = 14;
   var r = 8;
   var dkLen = 32;
-  var interruptStep = 200;
+  var interruptStep = 2000;
 
   var cb = function(derKey) {
     var err = null
